### ISSUE TITLE
False positive for E0401 (import-error) when running via setuptools and pytest

### ISSUE THREAD

**Issue by scranen** [PyCQA/pylint#2679]
### Steps to reproduce
1. Unzip [importerror.zip](https://github.com/PyCQA/pylint/files/2733560/importerror.zip) to an empty directory
2. In a clean python environment, run `python3 setup.py test`
3. Run `python3 setup.py test` again in the same environment

### Current behavior
In step 2, the PyLint tests fail with `Unable to import 'mimetype_match' (import-error)`
In step 3, the PyLint tests succeed
NB. the `test_pylint_version` test always fails, and is supposed to.

### Expected behavior
In step 2, the PyLint tests should also succeed. (The `test_pylint_version` should still fail.)

### pylint --version output
As generated by the test script (because pylint is downloaded by setuptools):
```
setup.py 2.2.2
astroid 2.1.0
Python 3.6.4 (default, Mar  9 2018, 23:15:03) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### Additional info
What I have been able to pin down so far is that the `mimetype_match` package that I used in my sample is a bit special in the sense that setuptools
- downloads a source release for it, then constructs a zipped `.egg`, 
- places it in the local `.eggs` directory and then 
- runs pytest.

The second time you run `setup.py`, the `.egg` is already in the `.eggs` directory, which probably makes the difference.

The reason I think the issue is with PyLint, is that the other tests _can_ import the package without issues.

**Comment by PCManticore** [comment]
@scranen Thanks for the detailed issue! I think this is mostly at an edge case, there are quite a lot of components involved in reproducing the issue. I think it should be expected to provide to `pylint` the modules it needs before it runs. Sorry for not being more helpful but I don't think this is that important to fix 🤷‍♂️
